hackathons/usoc23: Add USoC'23 landing page and Registration Challenges

StefanJum opened the pull request.

craciunoiuc commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+
+From zero-to-hero the two week event holds a number of starter tutorials, workshops and lessons on how to get build Unikraft unikernels for production systems.
+
+The workshop will be hands-on and will take place for 10 days, between August 29 and September 9, 2022, 4pm-8pm CEST, in English, virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
```
```suggestion
The workshop will be hands-on and will take place for 10 days, between August 29 and September 9, 2022, 4pm-7pm CEST, in English, virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
```

craciunoiuc commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+The workshop will be hands-on and will take place for 10 days, between August 29 and September 9, 2022, 4pm-8pm CEST, in English, virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
+Topics include building unikernels, benchmarking, debugging, porting applications, virtualization and platform specifics.
+The workshop will take place for two weeks (July 03 - July 16, 4pm-7pm CEST) virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
+There will be 6 3hrs sessions, followed by a full day hackathon will be held live at on `TBA` at [University POLITEHNICA of Bucharest](https://upb.ro), with the possibility to also join online via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
```
```suggestion
There will be six 3-hour sessions, followed by a full day hackathon which will be held live at in `TBA` at [University POLITEHNICA of Bucharest](https://upb.ro), with the possibility to also join online via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
```

craciunoiuc commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+To take part in the event, please fill this [registration form](https://forms.gle/ZUVLnkb6YNWhr9Cp8).
+You will need to complete a set of challenges that will get you accustomed to the environment you will be using during the sessions.
+
+It's recommended you checked these prerequisites before taking part in USoC'23:
```
```suggestion
It's recommended you check these prerequisites before taking part in USoC'23:
```

craciunoiuc commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+
+* fair knowledge of Linux command-line interface
+* fair knowledge of programming concepts and the C programming language
+* basic understanding of operating system concepts: process, threads, virtual memory, filesystems, file descriptors
```
```suggestion
* basic understanding of operating system concepts: processes, threads, virtual memory, filesystems, file descriptors
```

craciunoiuc commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+* fair knowledge of Linux command-line interface
+* fair knowledge of programming concepts and the C programming language
+* basic understanding of operating system concepts: process, threads, virtual memory, filesystems, file descriptors
+* some exposure to assembly language and computer architecture
```
```suggestion
* some exposure to the assembly language and computer architecture
```

craciunoiuc commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+
+### People
+
+USoC'23 will be held by members of the Unikraft community including professors, lecturers, PhD and MSci students from [University POLITEHNICA of Bucharest](https://upb.ro/) and the comercial side of Unikraft, [`Unikraft.io`](https://unikraft.io/).
```
Maybe there will be more universities? Will need to update after the list of volunteers is done

StefanJum replied:
Yes, we will update this on the go if it is the case.

craciunoiuc commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+
+USoC'23 consists of 6 sessions in 6 days and a hackathon.
+Each session is 3 hours long and consists of practical demos and then exercises for participants.
+The hackathon is a full day event where you'll be provided tasks and challenges that require you to enable, test, fix, evaluate and improve applications and libraries with Unikraft.
```
Also specify how many hours `10 hours` I think

StefanJum replied:
I will put 10hrs now for reference, and will update it after we decide the exact date, location and any other details.

craciunoiuc reviewed (changes requested):
Some comments from my side, looks good overall

RaduNichita commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+
+From zero-to-hero the two week event holds a number of starter tutorials, workshops and lessons on how to get build Unikraft unikernels for production systems.
+
+The workshop will be hands-on and will take place for 10 days, between August 29 and September 9, 2022, 4pm-7pm CEST, in English, virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
```
```suggestion
The workshop will be hands-on and will take place for 10 days, between August 29 and September 9, 2022, 4pm-7pm CEST, in English, virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
```

Maybe put TODO on the dates for now?

StefanJum replied:
We know the dates, I just missed them here.

RaduNichita commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+
+It's recommended you check these prerequisites before taking part in USoC'23:
+
+* fair knowledge of Linux command-line interface
```
```suggestion
* good knowledge of Linux command-line interface
```

StefanJum replied:
I would leave this to `fair`, Linux cli is not that strong of an item within Unikraft.

RaduNichita commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+It's recommended you check these prerequisites before taking part in USoC'23:
+
+* fair knowledge of Linux command-line interface
+* fair knowledge of programming concepts and the C programming language
```
```suggestion
* good knowledge of programming concepts and the C programming language
```

RaduNichita commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+
+### Schedule
+
+USoC'23 consists of 6 sessions in 6 days and a hackathon.
```
```suggestion
USoC'23 consists of 6 sessions and a hackathon.
```

RaduNichita commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+date: 2023-04-12T01:10:00+02:00
+draft: false
+collapsible: true
+description: Unikraft Summer of Code 2023, July 3 - 15, 2023
```
```suggestion
description: Unikraft Summer of Code 2023, July 4 - 15, 2023
```

RaduNichita commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+
+USoC'23 consists of 6 sessions in 6 days and a hackathon.
+Each session is 3 hours long and consists of practical demos and then exercises for participants.
+The hackathon is a full day event (10 hours) where you'll be provided tasks and challenges that require you to enable, test, fix, evaluate and improve applications and libraries with Unikraft.
```
```suggestion
The hackathon is a full day event (10 hours) where you'll be provided various tasks and challenges that require you to enable, test, fix, evaluate and improve applications and libraries with Unikraft.
```

RaduNichita reviewed (changes requested):
Nice job, @StefanJum ! :100: 

I left out a few nitpicks!!

razvand commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+
+Unikraft Summer of Code 2023 (*USoC'23*) is a **free and virtual** unikernel and library operating systems workshop held by members of the Unikraft community.
+
+From zero-to-hero the two week event holds a number of starter tutorials, workshops and lessons on how to get build Unikraft unikernels for production systems.
```
```suggestion
The two week event holds a number of starter tutorials, workshops and lessons on how to get build Unikraft unikernels for production systems.
```

razvand commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+
+From zero-to-hero the two week event holds a number of starter tutorials, workshops and lessons on how to get build Unikraft unikernels for production systems.
+
+The workshop will be hands-on and will take place for 10 days, between July 4 and July 15, 2023, 4pm-7pm CEST, in English, virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
```
```suggestion
The workshop will be hands-on and will take place for six days, between July 4 and July 15, 2023, 4pm-7pm CEST, in English, virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
```

razvand reviewed (changes requested):
Thanks, @StefanJum, this looks good. I made some small comments.

Please also see the linter warnings.

StefanJum commented:
> Thanks, @StefanJum, this looks good. I made some small comments.
> 
> Please also see the linter warnings.

I fixed the comments. The linter complains about the inner `html` within tables, but it's necessary, and it sees the dots in the dates (again, within the tables) as sentence ends (so `| ... 04.07.2023 ...|` is seen as more than one sentence per line).
We need to improve the linter to know when it is parsing tables, I'll open an issue on it.

I also added a commit with the challenges description, we will need to add a link to the `zip` archive with the files once we have it published.

RaduNichita commented on content/en/community/hackathons/usoc23/registration-challenges/_index.md:
```diff
@@ -0,0 +1,17 @@
+
+## Unikraft Summer of Code 2023 - Registration Challenges
+
+[Unikraft Summer of Code](community/hackathons/usoc23/) is a unikernel and library Operating Systems workshop held by members of the Unikraft community including professors, lecturers and PhD students.
```
```suggestion
[Unikraft Summer of Code](community/hackathons/usoc23/) is a unikernel and library Operating Systems workshop held by members of the Unikraft community including professors, lecturers and PhD and MSci students.
```

RaduNichita reviewed (changes requested):
A few more nitpicks, @StefanJum

RaduNichita commented on content/en/community/hackathons/sessions/registration-challenges/work/02-breaking-stuff/README.md:
````diff
@@ -0,0 +1,16 @@
+If everything goes well, the output of the executable will look like this:
+
+```console
+$ ./main
````
```suggestion
$ ./destroy-me
```

RaduNichita reviewed (changes requested):
One more nitpick from my side. It looks very good.

StefanJum commented:
Done @RaduNichita, thanks.

razvand commented on content/en/community/hackathons/sessions/registration-challenges/work/01-build-system/src/key.c:
```diff
@@ -0,0 +1,20 @@
+extern char *obf_string;
+static char plain_string[PLAIN_SIZE];
+
+static void deobfuscate(char *buffer, int len, char *key, int key_len)
```
Use Tab-based indentation.

razvand commented on content/en/community/hackathons/usoc23/registration-challenges/_index.md:
```diff
@@ -0,0 +1,17 @@
+
+## Unikraft Summer of Code 2023 - Registration Challenges
+
+[Unikraft Summer of Code](community/hackathons/usoc23/) is a unikernel and library Operating Systems workshop held by members of the Unikraft community including professors, lecturers and PhD and MSci students.
```
```suggestion
[Unikraft Summer of Code](community/hackathons/usoc23/) is a unikernel and libOS workshop held by members of the Unikraft community including professors and students.
```

razvand reviewed (changes requested):
First round. Need to also build and test it.

razvand commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+
+### People
+
+USoC'23 will be held by members of the Unikraft community including professors, lecturers, PhD and MSci students from [University POLITEHNICA of Bucharest](https://upb.ro/) and the comercial side of Unikraft, [`Unikraft.io`](https://unikraft.io/).
```
```suggestion
USoC'23 will be held by members of the Unikraft community including professors and students from [University POLITEHNICA of Bucharest](https://upb.ro/) and the comercial side of Unikraft, [`Unikraft.io`](https://unikraft.io/).
```

razvand commented on content/en/community/hackathons/sessions/registration-challenges/content/02-breaking-stuff-Challenge-Description.md:
````diff
@@ -0,0 +1,12 @@
+
+**Hint: All you need to know is in the executable file.**
+
+If everything goes well, the output of the executable will look like this:
+
+```console
+$ ./main
+[...]
+USoC{...}
+```
````
This is not relevant for this challenge.

razvand commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+draft: false
+collapsible: true
+description: Unikraft Summer of Code 2023, July 4 - 15, 2023
+image: /assets/imgs/usoc23.png
```
Remove this image, until we have a poster ready.

razvand commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,56 @@
+
+The workshop will be hands-on and will take place for six days, between July 4 and July 15, 2023, 4pm-7pm CEST, in English, virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
+Topics include building unikernels, benchmarking, debugging, porting applications, virtualization and platform specifics.
+The workshop will take place for two weeks (July 04 - July 16, 4pm-7pm CEST) virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
```
This line needs to go away, it's already mentioned above.

razvand commented on content/en/community/hackathons/usoc23/_index.md:
```diff
@@ -0,0 +1,54 @@
+
+The workshop will be hands-on and will take place for six days, between July 4 and July 15, 2023, 4pm-7pm CEST, in English, virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
+Topics include building unikernels, benchmarking, debugging, porting applications, virtualization and platform specifics.
+The workshops will be followed by a full day hackathon which will be held live in `TBA` at [University POLITEHNICA of Bucharest](https://upb.ro), with the possibility to also join online via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
```
Update TBA

StefanJum replied:
Should it be room PR706?

razvand replied:
Mention it will be online (this is the default). For those how can make it, we will also organize the hackathon in room PR706.

StefanJum commented:
Done @razvand.

razvand reviewed (approved):
Thanks, @StefanJum 

Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

craciunoiuc reviewed (approved):
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
